internal/helm/helmchart_structure_exists.go

```diff
@@ -67,21 +68,46 @@ func (r *HelmStructureExists) Test() *txqualitychecks.QualityResult {
 		return &txqualitychecks.QualityResult{ErrorDescription: "Can't read Helm Charts at charts/."}
 	}
 
+	// Check if the files/dir structure exists.
 	missingFiles := []string{}
+	chartYamlFiles := []string{}
 	for _, hc := range helmCharts {
 		if hc.IsDir() {
-			tmpFilesStructure := []string{}
 			for _, fname := range helmStructureFiles {
-				tmpFilesStructure = append(tmpFilesStructure, mainDir+"/"+hc.Name()+"/"+fname)
+				fpath := filepath.Join(mainDir, hc.Name(), fname)
+				isMissing := filesystem.CheckMissingFiles([]string{fpath})
+				if fname == "Chart.yaml" && isMissing == nil {
+					chartYamlFiles = append(chartYamlFiles, fpath)
+				} else if isMissing != nil {
+					missingFiles = append(missingFiles, isMissing...)
+				}
 			}
-			missingFiles = append(missingFiles, filesystem.CheckMissingFiles(tmpFilesStructure)...)
 		}
 	}
 
-	if len(missingFiles) > 0 {
-		return &txqualitychecks.QualityResult{ErrorDescription: "Following files are missing: " + strings.Join(missingFiles, " ")}
+	// Validate Chart.yaml files found in Helm Charts.
```

Comment:
perhaps create a fucntion if you need to descripe what a code snippet is doing in a comment

Reply:
Function validateChart created.